agent/rpc/peering/service.go

```diff
@@ -158,9 +158,9 @@ func (s *Service) GenerateToken(
 	writeReq := pbpeering.PeeringWriteRequest{
 		Peering: &pbpeering.Peering{
 			Name: req.PeerName,
-
 			// TODO(peering): Normalize from ACL token once this endpoint is guarded by ACLs.
 			Partition: req.PartitionOrDefault(),
+			Meta:      req.Meta,
```

Comment:
Before accepting this metadata here and in the other endpoint we should apply a similar pre-flight validation like we do for config entry meta. You could slightly rename this method, export it and then repurpose it here:
https://github.com/hashicorp/consul/blob/main/agent/structs/config_entry.go#L1118

Reply:
Thanks for the call out; let me work that in ⌨️